File: algorithms/1_easy/D/02325_decode_the_message/solution.cpp
class Solution {
public:
    string decodeMessage(string key, string message) {
        array<char, 26> key_map{};

        char curr_char = 'a';
        for (const char &ch: key) {
            size_t index = (ch - 'a');

            if (ch != ' ' && key_map[index] == 0) {
                key_map[index] = curr_char;
                curr_char += 1;
            }
        }

        string res{};

        for (const char &ch: message) {
            if (ch == ' ') {
                res.push_back(ch);
            } else {
                res.push_back(key_map[static_cast<size_t>(ch - 'a')]);
            }
        }

        return res;
    }
};

File: algorithms/1_easy/D/02325_decode_the_message/solution.py
class Solution:
    def decodeMessage(self, key: str, message: str) -> str:
        key_map = [0] * 26

        curr_char = ord('a')
        for ch in key:

            index = ord(ch) - ord('a')

            if ch != ' ' and key_map[index] == 0:
                key_map[index] = chr(curr_char)
                curr_char += 1

        res = []

        for ch in message:
            if ch == ' ':
                res.append(ch)
            else:
                res.append(key_map[ord(ch) - ord('a')])

        return "".join(res)
